benchmarks/python/test_matmul.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+from nvfuser import FusionDefinition, DataType
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from .core import run_benchmark, clear_cuda_cache
+import torch
+
+import csv
+import os
+
+
+def matmul_fusion(fd: FusionDefinition, dtype: DataType) -> None:
+    # Decide contiguity based on layout
+    a = fd.define_tensor(
+        shape=[-1, -1], contiguity=[True, True], dtype=dtype, is_cpu=False
+    )
+    b = fd.define_tensor(
+        shape=[-1, -1], contiguity=[True, True], dtype=dtype, is_cpu=False
+    )
+    out = fd.ops.matmul(a, b)
+    fd.add_output(out)
+
+
+precision_types = {"H": torch.float16, "S": torch.float32, "T": torch.bfloat16}
+
+
+def load_matmul_problems():
+    with open(os.path.join(os.path.dirname(__file__), "matmul_problems.csv"), "r") as f:
+        reader = csv.reader(f)
+        next(reader, None)  # skip header row
+        return list((int(m), int(n), int(k), layout) for m, n, k, layout in reader)
+
+
+@pytest.mark.parametrize("config", load_matmul_problems())
```

Comment:
We might also want to add the `ids=` kwarg here so that the benchmark names have easily filterable names.